File: js/templates/reply-automation/src/store.ts
import type { Classification } from "./classify.js"

export interface ProcessedReply {
  emailId?: string
  eventType?: string
  leadEmail?: string
  classification: Classification
  acted: boolean
  at: string
}

const replies: ProcessedReply[] = []

export function record(reply: ProcessedReply): void {
  replies.push(reply)
}

export function recent(limit = 50): ProcessedReply[] {
  return replies.slice(-limit).reverse()
}

File: python/templates/reply-automation/app/store.py
from typing import Any, Dict, List

_replies: List[Dict[str, Any]] = []


def record(reply: Dict[str, Any]) -> None:
    _replies.append(reply)


def recent(limit: int = 50) -> List[Dict[str, Any]]:
    return list(reversed(_replies[-limit:]))
